azurerm/internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -0,0 +1,438 @@
+
+			"location": azure.SchemaLocation(),
+
+			"vm_size": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"vm_priority": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(machinelearningservices.SystemAssigned),
+							}, false),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"scale_settings": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_node_count": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"min_node_count": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"node_idle_time_before_scale_down": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"subnet_resource_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceComputeClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	mlComputeClient := meta.(*clients.Client).MachineLearning.MachineLearningComputeClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+
+	// Get Machine Learning Workspace Name and Resource Group from ID
+	unparsedWorkspaceID := d.Get("machine_learning_workspace_id").(string)
+
+	workspaceID, err := parse.WorkspaceID(unparsedWorkspaceID)
+	if err != nil {
+		return err
+	}
+
+	existing, err := mlComputeClient.Get(ctx, workspaceID.ResourceGroup, workspaceID.Name, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("error checking for existing Compute Cluster %q in Workspace %q (Resource Group %q): %s",
+				name, workspaceID.Name, workspaceID.ResourceGroup, err)
+		}
+	}
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_machine_learning_compute_cluster", *existing.ID)
+	}
+
+	// AML Compute Cluster configuration (not needed here presumably)
+	vmSize := d.Get("vm_size").(string)
+	vmPriority := d.Get("vm_priority").(string)
+
+	inputScaleSettings := d.Get("scale_settings").([]interface{})
+	scaleSettings := expandScaleSettings(inputScaleSettings)
```

Comment:
```suggestion
	scaleSettings := expandScaleSettings(d.Get("scale_settings").([]interface{}))
```
Same here, could probably be combined as...